test/test_padacioso.py

```
@@ -229,3 +229,100 @@ def test_add_padatious_wildcard_intent(self):
+    def test_normalize_whitespace_util(self):
+        from padacioso.bracket_expansion import normalize_whitespace
+        self.assertEqual(normalize_whitespace("hello  world"), "hello world")
+        self.assertEqual(normalize_whitespace("  hello   world  "), "hello world")
+        self.assertEqual(normalize_whitespace("one\ttwo\nthree"), "one two three")
+        self.assertEqual(normalize_whitespace("already fine"), "already fine")
+        self.assertEqual(normalize_whitespace(""), "")
+
+    def test_normalize_apostrophes_util(self):
+        from padacioso.bracket_expansion import normalize_apostrophes
+        # curly right single quote
+        self.assertEqual(normalize_apostrophes("what’s up"), "what's up")
```

Ruff (0.15.10)

[warning] String contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF001)